components/builder-db/src/models/package.rs

```diff
@@ -245,29 +239,40 @@ impl Package {
 
     pub fn get_latest(req: GetLatestPackage, conn: &PgConnection) -> QueryResult<Package> {
         Counter::DBCall.increment();
-        diesel::sql_query("select * from get_origin_package_latest_v7($1, $2, $3)")
-            .bind::<Array<Text>, _>(req.ident.parts())
-            .bind::<Text, _>(req.target)
-            .bind::<Array<PackageVisibilityMapping>, _>(req.visibility)
+        Self::all()
+            .filter(origin_packages::ident_array.contains(req.ident.parts()))
+            .filter(origin_packages::target.eq(req.target))
+            .filter(origin_packages::visibility.eq(any(req.visibility)))
+            .order(sql::<Package>(
+                "to_semver(ident_array[3]) desc, ident_array[4] desc",
+            )).limit(1)
             .get_result(conn)
     }
 
     pub fn create(package: NewPackage, conn: &PgConnection) -> QueryResult<Package> {
         Counter::DBCall.increment();
-        diesel::sql_query("SELECT * FROM insert_origin_package_v5($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)")
-            .bind::<BigInt,_>(package.origin_id)
-            .bind::<BigInt,_>(package.owner_id)
-            .bind::<Text,_>(package.name)
-            .bind::<Text,_>(package.ident)
-            .bind::<Text,_>(package.checksum)
-            .bind::<Text,_>(package.manifest)
-            .bind::<Text,_>(package.config)
-            .bind::<Text,_>(package.target)
-            .bind::<Array<Text>,_>(package.deps)
-            .bind::<Array<Text>,_>(package.tdeps)
-            .bind::<Array<Integer>,_>(package.exposes)
-            .bind::<PackageVisibilityMapping,_>(package.visibility)
-        .get_result(conn)
+        let package = match diesel::insert_into(origin_packages::table)
+            .values(&package)
+            .returning(ALL_COLUMNS)
+            .on_conflict_do_nothing()
+            .get_result::<Package>(conn)
+        {
+            Ok(pkg) => pkg,
+            Err(e) => {
+                debug!("PACKAGE ERROR: {:?}", e);
```

Comment:
Either remove debug or update text to be more descriptive - eg "Create package error: {:?}" ?